src/secret_sharing/replicated/malicious/additive_share.rs

```diff
@@ -23,16 +23,37 @@ use std::{
 use typenum::Unsigned;
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct AdditiveShare<V: SharedValue> {
+pub struct AdditiveShare<V: SharedValue + ExtendableField> {
     x: SemiHonestAdditiveShare<V>,
-    rx: SemiHonestAdditiveShare<V>,
+    rx: SemiHonestAdditiveShare<V::LargeFieldType>,
 }
 
-impl<V: SharedValue> SecretSharing<V> for AdditiveShare<V> {
+pub trait ExtendableField {
+    type LargeFieldType: Field;
+    fn get_induced_value(&self) -> Self::LargeFieldType;
+}
+
+impl ExtendableField for Fp32BitPrime {
+    type LargeFieldType = Fp32BitPrime;
+
+    fn get_induced_value(&self) -> Self::LargeFieldType {
+        *self
+    }
+}
```

Comment:
```rust
impl<T> ExtendableField for T where T: PrimeField {
    type ExtendedField = Self;
    fn to_extended(&self) -> Self::ExtendedField { *self }
}
```

I know that Fp31 is too small here, but maybe we can address that in other ways, like sticking a `#[cfg(test)]` on it so that it never gets used in anger.

Otherwise, this is just another thing we need to fix if we go to a different field size.

What we might want is some way to make compilation conditional on the size of the prime, but we don't get to have that (yet).

Reply:
So I currently do have this configured to work for Fp31 (only in tests). I used to have that up here, but clippy was all mad, claiming that the import of Fp31 was "unused" due to it being wrapped in the #[cfg(test)]. I moved it down into the tests module to make that error go away.

What are you suggesting I change this to? Where would the #[cfg(test)] part go in the code you've shown above (defining it on PrimeField itself)?